src/distilabel/pipeline/base.py

```diff
@@ -247,9 +247,16 @@ def _set_runtime_parameters(self, parameters: Dict[str, Dict[str, Any]]) -> None
             parameters: A dictionary with the step name as the key and a dictionary with
             the parameter name as the key and the parameter value as the value.
         """
+        step_names = set(self.dag.G)
         for step_name, step_parameters in parameters.items():
-            step: "_Step" = self.dag.get_step(step_name)["step"]
-            step.set_runtime_parameters(step_parameters)
+            if step_name not in step_names:
+                self._logger.warning(
```

Comment:
I would be more specific here: "❓ Step '{step_name}' provided in `Pipeline.run(parameters={...})` not found in the pipeline..." or something similar